src/controller/CHIPDevice.cpp

```diff
@@ -179,12 +186,25 @@ void Device::OnNewConnection(SecureSessionHandle session, SecureSessionMgr * mgr
 {
     mState         = ConnectionState::SecureConnected;
     mSecureSession = session;
+#ifdef CHIP_APP_USE_INTERACTION_MODEL
+    if (mCommandSender != nullptr)
+    {
+        mCommandSender->Shutdown();
+        mCommandSender = nullptr;
+    }
+    chip::app::InteractionModelEngine::GetInstance()->NewCommandSender(&mCommandSender);
+#endif
```

Comment:
Since the other cases where `mCommandSender` is checked against `nullptr` do no wrap it with an #ifdef, I guess you can do:
```
    if (mCommandSender != nullptr)
    {
         mCommandSender->Shutdown();
         mCommandSender = nullptr;
    }
#ifdef CHIP_APP_USE_INTERACTION_MODEL
    chip::app::InteractionModelEngine::GetInstance()->NewCommandSender(&mCommandSender);
#endif